### DISCUSSION TITLE
2.0.0-beta.1

### DISCUSSION THREAD

**Discussion by JohnNiang** [halo-dev/halo#2696] [General]
> **Warning**
> 当前版本为非正式发布版本，**无法从 `1.5` 和 `1.6` 直接升级到当前版本，也不建议在生产环境使用**，欢迎大家测试并反馈。
>
> 另外，在 Halo 2.0 正式发布的时候，我们会提供完整的升级指南。

自 [v2.0.0-alpha.4](https://github.com/halo-dev/halo/releases/tag/v2.0.0-alpha.4) 发布以来的变更如下：

## 问题修复

- 修复日志 while 循环导致 CPU 占用高的问题。 @guqing <https://github.com/halo-dev/halo/pull/2663>
- 修复主题预览时跳转页面没有携带预览参数的问题。 @guqing <https://github.com/halo-dev/halo/pull/2661>

## 主要特性

- 支持通过配置修改文章列表分页大小。 @guqing <https://github.com/halo-dev/halo/pull/2660>
- Console 端重构主题预览弹框，支持选择主题以及针对主题进行设置。 @ruibaby <https://github.com/halo-dev/console/pull/683>
- Console 端的菜单管理支持设置主菜单。 @guqing @ruibaby <https://github.com/halo-dev/halo/pull/2667> <https://github.com/halo-dev/console/pull/684>
- 主题端提供 `pluginFinder.available()` API 以获取具体插件的可用状态。 @guqing <https://github.com/halo-dev/halo/pull/2683>
- 提供文章全文搜索功能并支持搜索引擎扩展。 @JohnNiang <https://github.com/halo-dev/halo/pull/2675>

## 重要变更

- 主题端获取默认菜单的 API 由 `menuFinder.getDefault()` 改为 `menuFinder.getPrimary()`。 @guqing <https://github.com/halo-dev/halo/pull/2667>
- 优化 Console 端获取系统初始化状态的方式，解决切换路由会发生卡顿的问题。 @ruibaby <https://github.com/halo-dev/console/pull/686>
- 重构 Console 端编辑器的创建方式。 @ruibaby <https://github.com/halo-dev/console/pull/670>
- 重构 Console 端文章和自定义页面的发布流程。 @ruibaby <https://github.com/halo-dev/console/pull/685> <https://github.com/halo-dev/console/pull/690>
- 移除 Console 端表单中的动画效果。 @ruibaby <https://github.com/halo-dev/console/pull/687>
- 重构文章发布以解决创建与发布 API 几乎同时调用时无法成功发布文章的问题。 @guqing <https://github.com/halo-dev/halo/pull/2659>

## 依赖变更

- 升级 Spring Boot 至 3.0.0-RC2。 @JohnNiang <https://github.com/halo-dev/halo/pull/2692>

## 其他资源

- 适用于 2.0 alpha 的插件和主题可查阅：<https://github.com/halo-sigs/awesome-halo>

**Full Changelog**: https://github.com/halo-dev/halo/compare/v2.0.0-alpha.4...v2.0.0-beta.1

<hr /><em>This discussion was created from the release <a href='https://github.com/halo-dev/halo/releases/tag/v2.0.0-beta.1'>2.0.0-beta.1</a>.</em>

**Comment by ruibaby** [comment]
> 提供文章全文搜索功能并支持搜索引擎扩展。 @JohnNiang https://github.com/halo-dev/halo/pull/2675

主题端如果需要搜索模块，请安装 https://github.com/halo-sigs/plugin-search-widget

此外，[theme-earth](https://github.com/halo-sigs/theme-earth) 和 [theme-anatole](https://github.com/halo-sigs/theme-anatole) 已经为插件提供了适配，安装即可显示调起搜索框的按钮。

其他主题如果需要适配，请参考一下示例代码：

```html
<div
  th:if="${pluginFinder.available('PluginSearchWidget')}"
  onclick="SearchWidget.open()"
>
  搜索
</div>
```

**Comment by ql0202cocou** [comment]
这么快就B测了啊！

大的要来了……
